stdlib/public/Synchronization/Atomics/WordPair.swift

```diff
@@ -183,6 +183,18 @@ extension WordPair: Hashable {
   }
 }
 
+@available(SwiftStdlib 6.2, *)
+extension WordPair: Comparable {
+  // Note: This function can have a lower availability than the conformance
+  // itself because it's always emit into client.
+  @available(SwiftStdlib 6.0, *)
+  @_alwaysEmitIntoClient
+  @_transparent
+  public func <(lhs: WordPair, rhs: WordPair) -> Bool {
+    lhs.first < rhs.first || lhs.second < rhs.second
```

Comment:
Won't this results in:

```swift
let c4 = WordPair(first: 1, second: 2)
let c5 = WordPair(first: 2, second: 1)
expectTrue(c4 < c5)
expectTrue(c5 < c4)
```

Reply:
You would be correct, thanks!

Reply:
Since the rationale for implementing `Comparable` here mentions that the type is essentially a nominal counterpart to `(first, second)` and should have the same comparison behavior, one way to ensure that we avoid running into think-os and never diverge our behavior from that rationale is by actually delegating all the hard work to the tuple comparison operator: `(lhs.first, lhs.second) < (rhs.first, rhs.second)`.